core/src/main/java/utilcalc/core/reportGen/DepositSectionGenerator.java

```diff
@@ -0,0 +1,46 @@
+package utilcalc.core.reportGen;
+
+import static utilcalc.core.utils.Util.ensureNotNegativeBigDecimalValue;
+
+import java.math.BigDecimal;
+import java.util.List;
+import utilcalc.core.model.input.DepositsSectionInputs;
+import utilcalc.core.model.input.Payment;
+import utilcalc.core.model.output.Deposit;
+import utilcalc.core.model.output.DepositSection;
+import utilcalc.core.model.output.ReportSection;
+
+public final class DepositSectionGenerator {
+
+    public static ReportSection generateDepositSection(DepositsSectionInputs depositsInputs) {
+        String name = depositsInputs.name();
+        List<Payment> payments = depositsInputs.payments();
+        List<Deposit> deposits = mapPaymentListToDepositList(payments);
+
+        BigDecimal totalAmount =
+                deposits.stream().map(Deposit::amount).reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        return new DepositSection(name, totalAmount, deposits);
+    }
+
+    private static Deposit mapPaymentToDeposit(Payment payment) {
+        String description = payment.description();
+        BigDecimal count = payment.count();
+        BigDecimal unitAmount = payment.unitAmount();
+
+        ensureNotNegativeBigDecimalValue(count, "deposit");
+        ensureNotNegativeBigDecimalValue(unitAmount, "unitAmount");
```

Comment:
unitAmount bych dovolil zaporny pro pripad, kdyby majitel z nejakeho duvodu vracel najemnikovi zalohu. Sice se to jeste nestalo, ale to omezeni mi prijde zbytecne.